bigbio/hub/hub_repos/bioid/bioid.py

```diff
@@ -0,0 +1,423 @@
+
+            if article_id not in annotations:
+                annotations[article_id] = {}
+
+            figure = record["figure"]
+
+            if figure not in annotations:
+                annotations[article_id][figure] = []
+
+            annotations[article_id][figure].append(record)
+
+        return annotations
+
+    def load_data(self, data_dir: str) -> List[Dict]:
+        """
+        Compose text from BioC files with annotations from `annotations.csv`.
+        We load annotations from `annotations.csv`
+        becuase the one in the BioC xml files have offsets issues.
+        """
+
+        text_dir = os.path.join(data_dir, "BioIDtraining_2", "caption_bioc")
+        annotation_file = os.path.join(data_dir, "BioIDtraining_2", "annotations.csv")
+
+        annotations = self.load_annotations(path=annotation_file)
+
+        data = []
+
+        for file_name in os.listdir(text_dir):
+
+            if file_name.startswith(".") or not file_name.endswith(".xml"):
+                continue
+
+            collection = bioc.load(os.path.join(text_dir, file_name))
+
+            for document in collection.documents:
+
+                item = document.infons
+
+                assert (
+                    len(document.passages) == 1
+                ), "Document contains more than one passage (figure caption). This is not expected!"
+
+                passage = document.passages[0]
+
+                article_id = document.infons["pmc_id"]
+                figure = document.infons["sourcedata_figure_dir"]
+
+                try:
+                    passage.annotations = annotations[article_id][figure]
+                except KeyError:
+                    passage.annotations = []
+
+                item["passages"] = [
+                    {
+                        "text": passage.text,
+                        "annotations": passage.annotations,
+                        "offset": passage.offset,
+                    }
+                ]
+
+                data.append(item)
+
+        return data
+
+    def get_entity(self, normalization: str) -> Tuple[str, List[Dict]]:
+        """
+        Compile normalization information from annotation
+        """
+
+        db_name_ids = normalization.split(":")
+
+        db_ids = None
+
+        # ids from cellosaurus do not have db name
+        if len(db_name_ids) == 1:
+            db_name = "Cellosaurus"
+            db_ids = db_name_ids[0].split("|")
+        else:
+            # quirk
+            if db_name_ids[0] == "CVCL_6412|CL":
+                db_name = "Cellosaurus"
+                db_ids = ["CVCL_6412"]
+            else:
+                db_name = db_name_ids[0]
+                # db_name hints for entity type: skip if does not provide normalization
+                if db_name not in self.ENTITY_TYPES_NOT_NORMALIZAD:
+                    db_ids = [i.split(":")[1] for i in normalization.split("|")]
+
+        normalized = (
```

Comment:
A lot of entities are normalized to {"db_name": "Uberon", "db_id": "Uberon"} - that certainly a bug?

Reply:
Thanks for catching this! I fixed it.